test/test_utils/__init__.py

```diff
@@ -1049,7 +1049,7 @@ def parse_canary_images(framework, region, image_type):
     versions = []
     for v, inf_train in versions_counter.items():
         # Earlier versions of huggingface did not have inference, Graviton is only inference
-        if (inf_train["inf"] and inf_train["tr"]) or framework.startswith("huggingface") or use_graviton:
+        if (inf_train["inf"] and image_type == "inference") or (inf_train["tr"] and image_type == "training") or framework.startswith("huggingface") or use_graviton:
```

Comment:
As per my understanding, earlier, we used to iterate through all the tags to see if both  - `inf_train["inf"]` and `inf_train["tr"]` are set for any image that is non-HF and non-Graviton. For HF images and Graviton images, we only had training and inference images respectively. 

Now, we are changing the logic to ensure that even if one of the 2 i.e. `inf_train["inf"]` or `inf_train["tr"]` is set and the `image_type` is `inference` or `training` respectively, then, we run the canary for just the inference type of images or the training type of images - based on what is the image_type. Right?